### PR TITLE
Add support for CURLOPT_SSH_KNOWNHOSTS, CURLOPT_SSH_KEYFUNCTION

### PR DESCRIPTION
**Pull request by nojb**
This PR is an initial try at adding support for the [`CURLOPT_SSH_KNOWNHOSTS`](https://curl.haxx.se/libcurl/c/CURLOPT_SSH_KNOWNHOSTS.html) and [`CURLOPT_SSH_KEYFUNCTION`](https://curl.haxx.se/libcurl/c/CURLOPT_SSH_KEYFUNCTION.html) options.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -135,7 +135,7 @@ CURLOPT_PROXY_TRANSFER_MODE, CURLOPT_SEEKFUNCTION, CURLOPT_OPENSOCKETFUNCTION,
 CURLOPT_PROTOCOLS, CURLOPT_REDIR_PROTOCOLS, CURLOPT_RESOLVE, CURLOPT_DNS_SERVERS,
 CURLOPT_MAIL_FROM, CURLOPT_MAIL_RCPT, CURLOPT_PIPEWAIT, CURLOPT_CERTINFO,
 CURLOPT_USERNAME, CURLOPT_PASSWORD, CURLOPT_LOGIN_OPTIONS, CURLOPT_CONNECT_TO,
-CURLOPT_POSTREDIR, CURLOPT_MIMEPOST,
+CURLOPT_POSTREDIR, CURLOPT_MIMEPOST, CURLOPT_SSH_KNOWNHOSTS, CURLOPT_SSH_KEYFUNCTION,
```
[the remark this answers is not in the document]
Comment (nojb) [reply configure.ac]: OK, fixed.

**Comment by nojb** [comment]
Thanks for the review and the fixes (indeed, the PR code was a bit rushed, sorry for that!). I think I addressed the remaining issues: removed the configure check and added some docs.

**Comment by ygrek** [comment]
Thanks!

**Comment by nojb** [comment]
Thank you!

**Pull request merged**